koku/koku/middleware.py

```diff
@@ -371,13 +356,14 @@ def process_request(self, request):  # noqa: C901
                 DB_CONNECTION_ERRORS_COUNTER.inc()
                 return HttpResponseFailedDependency({"source": "Database", "exception": err})
 
+            user_key = f"{org_id}_{username}"
             try:
-                if username not in USER_CACHE:
-                    user = User.objects.get(username=username)
-                    USER_CACHE[username] = user
-                    LOG.debug(f"User added to cache: {username}")
+                if user_key not in USER_CACHE:
+                    user = IdentityHeaderMiddleware.create_user(username, email, customer, request)
```

Comment:
If we eliminate the `IdentityHeaderMiddleware.create_user` in the `_create_context` method in the `ProviderBuilder`, then I think we can just get rid of the `create_user` method entirely and just replace all of this with:
```suggestion
                    user = User(username=username, email=email, customer=customer)
```